src/features/copilotTools.ts

```diff
@@ -36,55 +50,181 @@ export class GetPackagesTool implements LanguageModelTool<IGetActiveFile> {
+            envInfo.version = environment.version;
+
+            // does this need to be refreshed prior to returning to get any new packages?
             await this.api.refreshPackages(environment);
             const installedPackages = await this.api.getPackages(environment);
-
-            let resultMessage: string;
             if (!installedPackages || installedPackages.length === 0) {
-                resultMessage = 'No packages are installed in the current environment.';
+                envInfo.packages = [];
             } else {
-                const packageNames = installedPackages
-                    .map((pkg) => (pkg.version ? `${pkg.name} (${pkg.version})` : pkg.name))
-                    .join(', ');
-                resultMessage = 'The packages installed in the current environment are as follows:\n' + packageNames;
+                envInfo.packages = installedPackages.map((pkg) =>
+                    pkg.version ? `${pkg.name} (${pkg.version})` : pkg.name,
+                );
             }
 
-            const textPart = new LanguageModelTextPart(resultMessage || '');
+            // format and return
+            const textPart = BuildEnvironmentInfoContent(envInfo);
             deferredReturn.resolve({ content: [textPart] });
         } catch (error) {
-            const errorMessage: string = `An error occurred while fetching packages: ${error}`;
+            const errorMessage: string = `An error occurred while fetching environment information: ${error}`;
             deferredReturn.resolve({ content: [new LanguageModelTextPart(errorMessage)] } as LanguageModelToolResult);
         }
         return deferredReturn.promise;
     }
-
     /**
      * Prepares the invocation of the tool.
      * @param _options - The preparation options.
      * @param _token - The cancellation token.
      * @returns The prepared tool invocation.
      */
     async prepareInvocation?(
-        _options: LanguageModelToolInvocationPrepareOptions<IGetActiveFile>,
+        _options: LanguageModelToolInvocationPrepareOptions<IResourceReference>,
         _token: CancellationToken,
     ): Promise<PreparedToolInvocation> {
-        const message = 'Preparing to fetch the list of installed Python packages...';
+        const message = 'Preparing to fetch Python environment information...';
+        return {
+            invocationMessage: message,
+        };
+    }
+}
+
+function BuildEnvironmentInfoContent(envInfo: EnvironmentInfo): LanguageModelTextPart {
+    // Create a formatted string that looks like JSON but preserves comments
+    const content = `{
+  // type of python environment; sys means it is the system python
+  "environmentType": ${JSON.stringify(envInfo.type)},
+  // python version of the environment
+  "pythonVersion": ${JSON.stringify(envInfo.version)},
+  // command to run python in this environment, will include command with active environment if applicable
```

Comment:
The description could be assertive:
"Use this command to run python script or code in the terminal"

I don't think there is need to specify activation.